src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -159,29 +172,37 @@ final case class RequestSession[F[_]: Monad](
   def flatMap[R, R1](r: F[R], f: R => F[R1]): F[R1] = r.flatMap(f)
 }
 
-class GenericClient[F[_]: Monad](
+class GenericClient[F[_]](
```

Comment:
Why did we go from `F[_]: Monad` to `F[_]()(implicit monad: Monad[F]` and the other changes to this class construction?

Reply:
If I remember correctly, it was done to be able to type `F.pure` instead of `Applicative[F].pure`, but we're not doing that anymore, so I think we can go back to a context bound. Good catch, thanks!

Reply:
The other changes are done to retain the previous constructor signature (which takes an Auth), while also adding support for AuthProvider[F].

I'm not super happy about the duplicate constructors/apply()s. It could probably have been made more elegant using some implicit conversions from Auth to AuthProvider or something, but I wanted to avoid users having to add any additional imports just to use the SDK as they would normally.